packages/tldraw/src/state/TLDR.ts

```diff
@@ -394,26 +395,46 @@ export class TLDR {
 
     const shapes = data.document.pages?.[data.appState.currentPageId].shapes
     const groupShape = shapes?.[ids[0]]
+    const isSingle = ids.length === 1
+
+    const mirror = (data: TDSnapshot, shapeId: string, pageId: string, index: number) => {
```

Comment:
ah I see now why I was so confused yesterday, it looks like this `mutateShapes` function _is_ supposed to be generic, and it shouldn't know anything about flipping shapes. All the flipping logic you've added here should really live in the `flipShapes.ts` file. That should simplify the logic quite a lot too. There might still need to be one or two small changes in this file to allow it to work properly (e.g. for treating children differently when the group is not the only selected shape) but the diff in this file should be minimal. Let me know if you want to pair on this, i'm happy to help :)